cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller.go

```diff
@@ -448,11 +450,10 @@ func (c *machineController) nodeGroupForNode(node *apiv1.Node) (*nodegroup, erro
 			if err != nil {
 				return nil, fmt.Errorf("failed to build nodegroup for node %q: %v", node.Name, err)
 			}
-			// We don't scale from 0 so nodes must belong
-			// to a nodegroup that has a scale size of at
-			// least 1.
-			if nodegroup.MaxSize()-nodegroup.MinSize() < 1 {
-				return nil, nil
+			if !nodegroup.scalableResource.CanScaleFromZero() {
+				if nodegroup.MaxSize()-nodegroup.MinSize() < 1 {
+					return nil, nil
+				}
```

Comment:
In case it's scale up from zero, you should also check if `nodegroup.MaxSize()-nodegroup.MinSize() >= 0`